### PR TITLE
HPCC-17845 Add logout button to common ESP pages and fix bugs

### PR DESCRIPTION
**Pull request by wangkx**
The common ESP pages are HTTTP test form and SOAP test form.
The logout button only displays if the esp is in a session
authentioncation mode. After the forms are logged out, a simple
logout page will be displayed.

Also fixed a bug when reading default unresticted resources.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck please review.

File: esp/xslt/gen_form.xsl [hunk 1]
```diff
@@ -104,9 +105,24 @@
                 <script type="text/javascript" src="files_/hashtable.js"/>
                 <script type="text/javascript" src="files_/gen_form.js"/>
                 <script type="text/javascript">         
-  
   var isIE = (navigator.appName == "Microsoft Internet Explorer");  
   
+  function logout()
```
Comment (afishbeck) [inline esp/xslt/gen_form.xsl]: Could this routine be put in a common javascript file?

**Comment by afishbeck** [comment]
@wangkx minor comment.  But you are also combining two completely unrelated changes in this commit.

**Comment by wangkx** [comment]
@afishbeck the logout() has been moved to gen_form.js and the title is changed. Please review.

File: esp/xslt/soap_page.xsl [hunk 2]
```diff
@@ -37,6 +39,23 @@
 
 <script type="text/javascript">
 var showhttp = '<xsl:value-of select="$showhttp"/>';
+
+  function logout()
```
Comment (afishbeck) [inline esp/xslt/soap_page.xsl]: @wangkx I was actually hoping this code would be somewhere shared.

**Comment by wangkx** [comment]
@afishbeck the logout() is moved into new logout.js which is shared by 2 xsl files.

**Review by afishbeck — APPROVED** [review]
@wangkx looks fine

**Pull request merged**